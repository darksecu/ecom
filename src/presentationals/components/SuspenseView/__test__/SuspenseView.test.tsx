import * as React from 'react';
import {View} from 'react-native';
import {SuspenseView} from '..';
import {fireEvent, render} from 'react-native-testing-library';

describe('Suspense view component', () => {
  it('Loading in Progress, Should Render Activity Indicator', () => {
    const tree = render(<SuspenseView isLoading={true} list={[]} />);

    expect(tree.findByTestId('SuspenseViewActivityIndicator')).toBeDefined();
  });

  it('Loading Done, Should Render Child', () => {
    const tree = render(
      <SuspenseView isLoading={true} list={[]}>
        <View />
      </SuspenseView>,
    );

    expect(tree).toMatchSnapshot();
  });
});
