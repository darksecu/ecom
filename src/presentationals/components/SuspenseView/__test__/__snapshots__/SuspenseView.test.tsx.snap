// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suspense view component Loading Done, Should Render Child 1`] = `<View />`;
